dotnet/src/Extensions/Planning.StepwisePlanner/StepwisePlanner.cs

```diff
@@ -359,28 +358,24 @@ private ChatHistory CreateChatHistory(IKernel kernel, out IAIService aiService)
         return chatHistory;
     }
 
-    private async Task<string> GetUserManualAsync(SKContext context)
+    private async Task<string> GetUserManualAsync(SKContext context, CancellationToken cancellationToken)
     {
         var descriptions = await this.GetFunctionDescriptionsAsync().ConfigureAwait(false);
         context.Variables.Set("functionDescriptions", descriptions);
-        return await this._promptRenderer.RenderAsync(this._manualTemplate, context).ConfigureAwait(false);
+        return await this._promptRenderer.RenderAsync(this._manualTemplate, context, cancellationToken).ConfigureAwait(false);
     }
 
-    private Task<string> GetUserQuestionAsync(SKContext context)
-    {
-        return this._promptRenderer.RenderAsync(this._questionTemplate, context);
-    }
+    private Task<string> GetUserQuestionAsync(SKContext context, CancellationToken cancellationToken)
+        => this._promptRenderer.RenderAsync(this._questionTemplate, context, cancellationToken);
 
-    private Task<string> GetSystemMessage(SKContext context)
-    {
-        return this._promptRenderer.RenderAsync(this._promptTemplate, context);
-    }
+    private Task<string> GetSystemMessageAsync(SKContext context, CancellationToken cancellationToken)
+        => this._promptRenderer.RenderAsync(this._promptTemplate, context, cancellationToken);
 
     #endregion setup helpers
 
     #region execution helpers
 
-    private Task<string> GetNextStepCompletion(List<SystemStep> stepsTaken, ChatHistory chatHistory, IAIService aiService, int startingMessageCount, CancellationToken token)
+    private Task<string> GetNextStepCompletionAsync(List<SystemStep> stepsTaken, ChatHistory chatHistory, IAIService aiService, int startingMessageCount, CancellationToken token)
```

Comment:
Why aren't these kind of rules getting flagged any longer??? Concerned there's a big pile of stuff we're missing now.

Reply:
Not sure. I'll check the rules.

Reply:
Most likely because there's no 'async' keyword in the method signature.